magento/pwa-studio
# Swap React Head to React Helmet Async

## Description

This PR changes the underlying react-head package to react-helmet-async instead. 

## Related Issue





Closes #2414 
Maybe #1719

## Acceptance 
### Verification Stakeholders
@jimbo 
@supernova-at 
@revanth0212 

### Specification



### Verification Steps


1. Go to the Homepage
2. Verify the title tag is `Home Page - ${STORE_NAME}`
3. Go to a product page
4. Verify the title tag is `${product.name} - ${STORE_NAME}`
5. Verify the Meta tag `<Meta name="description" content={product.meta_description} />` is present
6. Add a product and go to the cart page
7. Verify the title tag is {`Cart - ${STORE_NAME}`

## Comments
I tried to keep with the existing implementation as much as possible, hence keeping the implementation of Link, Meta, Title and Style. Going forward these could get removed in favour of using Helmet directly, ie
```
<Helmet>
   <title>Page Title</title>
</Helmet>
```
I also added `Helmet.defaultProps.defer = false;` as I believe this is closest to the current but this could go.

I tried unsuccessfully to keep all the existing tests as is. I believe this might be due to the portal approach previously used but could need some suggestions there.

One behaviour change is how an empty HeadProvider is handled
HeadProvider should delete all title tags if any
to
HeadProvider should keep existing title tags if any
as I believe the previous behaviour was mainly due to how react-head worked.

## Screenshots / Screen Captures (if appropriate)
![image](https://user-images.githubusercontent.com/455508/82429308-c616f800-9adf-11ea-9b75-0508dbb9689e.png)

## Checklist

* I have updated the documentation accordingly, if necessary.
[x] I have added tests to cover my changes, if necessary.
